Source/WTF/wtf/FixedBitVector.h

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/Assertions.h>
+#include <wtf/Atomics.h>
+#include <wtf/HashTraits.h>
+#include <wtf/PrintStream.h>
+#include <wtf/StdIntExtras.h>
+#include <wtf/StdLibExtras.h>
+
+namespace WTF {
+
+DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(FixedBitVector);
+DEFINE_ALLOCATOR_WITH_HEAP_IDENTIFIER(BitVector);
+
+class FixedBitVector final {
+    WTF_MAKE_FAST_ALLOCATED;
+    using WordType = UCPURegister;
+
+public:
+    FixedBitVector() = default;
+
+    ~FixedBitVector()
+    {
+        if (m_bits)
+            FixedBitVectorMalloc::free(m_bits);
+    }
+
+    bool concurrentAllocateOnce(size_t);
+    bool concurrentTest(size_t bitIndex, Dependency = Dependency());
+    bool concurrentTestAndSet(size_t bitIndex, Dependency = Dependency());
+
+    void allocateOnce(size_t);
```

Comment:
Let's not modify the size after it is constructed (so let's not add allocate method)
We should take size in the constructor, and we should use it like "vector = FixedBitVector(42)"

Reply:
Since we use `BitVector` as the storage. Maybe we can do
```
void initializeReferencedFunctionsTracker() const { m_referencedFunctions.resize(functionIndexSpaceSize()); }
```
with
```
void FixedBitVector::resize(size_t size) { m_bitVector.resize(size); }
```